polyfill/lib/ecmascript.mjs

```diff
@@ -3187,33 +3176,30 @@ export const ES = ObjectAssign({}, ES2020, {
     roundingMode,
     dayLengthNs = 86400e9
   ) => {
-    let quantity = 0;
+    let quantity = bigInt.zero;
     switch (unit) {
       case 'day':
-        quantity =
-          ((((hour * 60 + minute) * 60 + second) * 1000 + millisecond) * 1000 + microsecond) * 1000 + nanosecond;
-        quantity /= dayLengthNs;
-        break;
       case 'hour':
-        quantity = ((second + millisecond * 1e-3 + microsecond * 1e-6 + nanosecond * 1e-9) / 60 + minute) / 60 + hour;
-        break;
+        quantity = bigInt(hour);
+      // fall through
       case 'minute':
-        quantity = (second + millisecond * 1e-3 + microsecond * 1e-6 + nanosecond * 1e-9) / 60 + minute;
-        break;
+        quantity = quantity.multiply(60).plus(minute);
+      // fall through
       case 'second':
-        quantity = second + millisecond * 1e-3 + microsecond * 1e-6 + nanosecond * 1e-9;
-        break;
+        quantity = quantity.multiply(60).plus(second);
+      // fall through
       case 'millisecond':
-        quantity = millisecond + microsecond * 1e-3 + nanosecond * 1e-9;
-        break;
+        quantity = quantity.multiply(1000).plus(millisecond);
+      // fall through
       case 'microsecond':
-        quantity = microsecond + nanosecond * 1e-3;
-        break;
+        quantity = quantity.multiply(1000).plus(microsecond);
+      // fall through
       case 'nanosecond':
-        quantity = nanosecond;
-        break;
+        quantity = quantity.multiply(1000).plus(nanosecond);
     }
-    const result = ES.RoundNumberToIncrement(quantity, increment, roundingMode);
+    const nsPerUnit = unit === 'day' ? dayLengthNs : nsPerTimeUnit[unit];
```

Comment:
Could add 'day' to the table.

Reply:
My understanding was that (for DST reasons) dayLengthNs could vary depending on the caller's parameters, so I left it out of the table. LMK if this is a problem.